distributed/deploy/spec.py

```diff
@@ -333,7 +333,7 @@ def _correct_state(self):
                 await asyncio.wait(workers)
                 for w in workers:
                     w._cluster = weakref.ref(self)
-                    await w  # for tornado gen.coroutine support
+                await asyncio.gather(*workers)
```

Comment:
@jcrist, mentioned in [this comment]( https://github.com/rapidsai/dask-cuda/issues/198#issuecomment-561393289 ) that we should use `return_exceptions=True`. Though it sounds like `asyncio.gather` will just raise exceptions if we don't pass that argument, which seems similar to the previous behavior (excepting order is not guaranteed). So IDK if this is good enough or if we still need to handle exceptions for some other reason. If so, please advise on what sort of handling is expected. Thanks!

Reply:
Actually, I missed the `asyncio.wait` call above here, which should already successfully wait on all workers. Can we just delete the `await w` call entirely and not replace it with `asyncio.gather`? I don't see why we'd need to individually wait each worker in the case of tornado `gen.coroutine`, the `asyncio.wait` call should handle those equally fine.

Reply:
Sure happy to give that a try. The comment worried me, but I try your judgement here more than my own 🙂

Reply:
@mrocklin, do you know why the `await` was needed here? I see it was added in PR ( https://github.com/dask/distributed/pull/2675 ). Maybe [this discussion]( https://github.com/dask/distributed/pull/2675#discussion_r286161172 ) is relevant?

Reply:
FWIW I dug a little further and it looks `asyncio.gather` may not have always worked with Tornado, but they did fix it and start testing with `asyncio.gather` in 5.0.0 (see [this line]( https://github.com/tornadoweb/tornado/pull/2253/files#diff-fe992c84ff138a2e164479c6b81d4668R1074 )). So this should be fine.